sdk/identity/azure-identity/src/main/java/com/azure/identity/UsernamePasswordCredentialBuilder.java

```diff
@@ -36,6 +36,31 @@ public UsernamePasswordCredentialBuilder password(String password) {
         return this;
     }
 
+    /**
+     * Sets whether to use an unprotected file specified by <code>cacheFileLocation()</code> instead of
+     * Gnome keyring on Linux. This is false by default.
+     *
+     * @param allowUnencryptedCache whether to use an unprotected file for cache storage.
+     *
+     * @return An updated instance of this builder with the unprotected token cache setting set as specified.
+     */
+    public UsernamePasswordCredentialBuilder allowUnencryptedCache(boolean allowUnencryptedCache) {
+        this.identityClientOptions.allowUnencryptedCache(allowUnencryptedCache);
+        return this;
+    }
+
+    /**
+     * Sets whether to enable using the shared token cache. This is disabled by default.
+     *
+     * @param enabled whether to enabled using the shared token cache.
+     *
+     * @return An updated instance of this builder with if the shared token cache enabled specified.
+     */
+    public UsernamePasswordCredentialBuilder enableSharedTokenCache(boolean enabled) {
```

Comment:
We hadn't added the ability to use the shared token cache to the `UsernamePasswordCredential` in .NET and Python yet. I'm not totally against it, but I think it adds limited value since `UsernamePasswordCredential` has no user interaction.

Reply:
we'll be keeping it for preview, as discussed.